compose/foundation/foundation/src/desktopMain/kotlin/androidx/compose/foundation/internal/ClipboardUtils.desktop.kt

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2025 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.foundation.internal
+
+import androidx.compose.ui.platform.ClipEntry
+import androidx.compose.ui.platform.NativeClipboard
+import androidx.compose.ui.text.AnnotatedString
+import java.awt.datatransfer.DataFlavor
+import java.awt.datatransfer.StringSelection
+import java.io.IOException
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+internal actual suspend fun ClipEntry.readText(): String? {
+    if (!hasText()) return null
+
+    return withContext(Dispatchers.IO) {
+        try {
+            transferable.getTransferData(DataFlavor.stringFlavor) as? String
+        } catch (_: IOException) {
+            // the data is no longer available in the requested flavor
+            null
+        }
+    }
+}
+
+// TODO: https://youtrack.jetbrains.com/issue/CMP-7543/Support-styled-text-in-PlatformClipboard-implementations
+internal actual suspend fun ClipEntry.readAnnotatedString(): AnnotatedString? {
+    return readText()?.let { AnnotatedString(it) }
+}
+
+// TODO: https://youtrack.jetbrains.com/issue/CMP-7543/Support-styled-text-in-PlatformClipboard-implementations
+internal actual fun AnnotatedString?.toClipEntry(): ClipEntry? {
+    if (this == null) return null
+    val transferable = StringSelection(this.text)
+    return ClipEntry(transferable)
+}
+
+internal actual fun ClipEntry?.hasText(): Boolean {
+    if (this == null) return false
+    return transferable.isDataFlavorSupported(DataFlavor.stringFlavor)
+}
+
+// Here we rely on the NativeClipboard directly instead of using ClipEntry,
+// because getClipEntry is a suspend function, but in ContextMenu.desktop.kt we have older code
+// expecting a synchronous execution.
+internal fun NativeClipboard.blockingHasText(): Boolean {
```

Comment:
I don't think it needs to be named `blockingXXX`, but I don't feel that strongly about it.

Reply:
Since NativeClipboard is actualized as Any, it's not possible to name it `hasText` - it creates a conflict with `internal actual fun ClipEntry?.hasText(): Boolean `

Reply:
`ClipEntry?.hasText()` is not suspend too. So "blocking" is not really a feature that distingush one function from another. 

I choose another name - `nativeClipboardHasText`